network/tools/network-policy-enforcement-latency/pkg/client.go

```diff
@@ -0,0 +1,548 @@
+func (c *TestClient) initialize() error {
+	if err := c.verifyConfig(); err != nil {
+		return fmt.Errorf("failed to verify configuration, error: %v", err)
+	}
+
+	k8sClient, err := utils.NewK8sClient()
+	if err != nil {
+		return fmt.Errorf("failed to create k8s client, error: %v", err)
+	}
+
+	c.k8sClient = k8sClient
+	c.podCreationWorkQueue = workqueue.New()
+
+	return nil
+}
+
+// verifyConfig checks that all the mandatory test client config fields are
+// populated accordingly.
+func (c *TestClient) verifyConfig() error {
+	if len(c.config.testClientNamespace) == 0 {
+		return fmt.Errorf("testClientNamespace parameter is not specified")
+	}
+
+	if len(c.config.targetLabelSelector) == 0 {
+		return fmt.Errorf("targetLabelSelector parameter is not specified")
+	}
+
+	if len(c.config.targetNamespace) == 0 {
+		return fmt.Errorf("targetNamespace parameter is not specified")
+	}
+
+	if c.config.targetPort == 0 {
+		return fmt.Errorf("targetPort parameter is not specified")
+	}
+
+	if c.config.workerCount < 1 {
+		c.config.workerCount = utils.DefaultWorkerCount
+		klog.Infof("workerCount value is less than 1. The workerCount value is set to the default (%d)", utils.DefaultWorkerCount)
+	} else if c.config.workerCount > utils.MaxWorkerCount {
+		c.config.workerCount = utils.MaxWorkerCount
+		klog.Infof("workerCount value is higher than the maximum allowed. The workerCount value is set to the maximum (%d)", utils.MaxWorkerCount)
+	}
+
+	testTypeName := "pod-creation-enforcement-latency"
+	if !c.config.testPodCreation {
+		if len(c.config.allowPolicyName) == 0 {
+			return fmt.Errorf("allowPolicyName is not specified for policy creation test")
+		}
+		testTypeName = "policy-creation-enforcement-latency"
+	}
+
+	klog.Infof("Connectivity test client mode: %q", testTypeName)
+
+	// Log all test client config flags.
+	klog.Infof("Network policy enforcement latency test client parameters:")
+	klog.Infof("--testClientNamespace=%s", c.config.testClientNamespace)
+	klog.Infof("--targetLabelSelector=%s", c.config.targetLabelSelector)
+	klog.Infof("--targetNamespace=%s", c.config.targetNamespace)
+	klog.Infof("--targetPort=%d", c.config.targetPort)
+	klog.Infof("--expectedTargets=%d", c.config.expectedTargets)
+	klog.Infof("--allowPolicyName=%s", c.config.allowPolicyName)
+	klog.Infof("--metricsPort=%d", c.config.metricsPort)
+	klog.Infof("--testPodCreation=%t", c.config.testPodCreation)
+	klog.Infof("--workerCount=%d", c.config.workerCount)
+
+	return nil
+}
+
+// measureNetPolicyCreation runs the network policy enforcement latency test for
+// network policy creation.
+func (c *TestClient) measureNetPolicyCreation() error {
+	klog.Infof("Starting to measure network policy enforcement latency on network policy creation")
+	podList, err := c.targetPodList()
+	if err != nil {
+		return fmt.Errorf("failed to get the pod list, error: %v", err)
+	}
+	klog.Infof("%d pods listed", len(podList))
+
+	c.policyCreatedTime = &timeWithLock{lock: sync.RWMutex{}}
```

Comment:
Why is it intialized here, not in the constructor?

Reply:
The test client runs in one of the modes, and it instantiates objects for the specific mode once that mode is starting to run.
I think it's better to instantiate it closer to use, rather than doing it early for all variables, even the ones not to be used.

Reply:
Fixed